### ISSUE TITLE
Typescript typings

### ISSUE THREAD

**Issue by orefalo** [aarondcohen/id128#12]
would be nice.

**Comment by aarondcohen** [comment]
I don't personally use typescript, however, if you'd like to create a branch with the typings, i'd be happy to review them.

**Comment by orefalo** [comment]
Sorry got distracted

I believe that's it.

Put this file at the top -

file: index.d.ts
```
export declare namespace id128 {

    export interface Ulid {
        MIN(): Ulid;
        MAX(): Ulid;

        compare(rhs: Ulid): number;
        equal(rhs: Ulid): boolean;
        clone(): Ulid;
        time(): Ulid;

        bytes(): Int8Array;
        construct(bytes: Int8Array): Ulid;
        generate(): Ulid;

        fromCanonical(canonical: string): Ulid;
        fromCanonicalTrusted(canonical: string): Ulid;
        fromRaw(raw: string): Ulid;
        fromRawTrusted(raw: string): Ulid;

        toCanonical(): string;
        toRaw(): string;
    }
}
```

Change package.json as follows:
```
...
    "main": "index.js",
    "types": "index.d.ts",
	"files": [
        "index.js",
        "index.d.ts",
		"utils.js",
		"src/"
	],
...
```

republish - should work

**Comment by aarondcohen** [comment]
i'll cut a branch with this and the other related definitions this weekend.

**Comment by orefalo** [comment]
tks,

On thing I noticed, it the way you pass parameters using {time}.  That's not trivial to type with TS, and the reason I removed it.

**Comment by aarondcohen** [comment]
i got distracted.  this is still coming

**Comment by aarondcohen** [comment]
This is long overdue, but lockdown is the mother of invention: https://github.com/aarondcohen/id128/pull/20

Please try it out and log any issues on the PR.

**Comment by aarondcohen** [comment]
Bindings have been merged and published with version 1.6.0.  Will keep this open for 1 week for feedback.
